src/syscall/syscall_unix.go

```diff
@@ -16,3 +21,54 @@ type SockaddrInet6 struct {
 	Addr   [16]byte
 	raw    RawSockaddrInet6
 }
+
+func Fork() (err error) {
+	fail := int(libc_fork())
+	if fail < 0 {
+		// TODO: parse the syscall return codes
+		return errors.New("fork failed")
```

Comment:
if you can wrap some standard error in here, tests are easier.
syscall.EAGAIN is at least as good as this errors.New I think?